repair.lic

```diff
@@ -0,0 +1,227 @@
+    @town           = args.town.capitalize || UserVars.last_repair_town || @settings.hometown
+    @repair_timer   = @settings.repair_timer
+    @sort_head      = @settings.sort_auto_head
+    @disciplines    = ['forging', 'tinkering', 'carving', 'shaping', 'outfitting', 'alchemy', 'enchanting']
+    @toolbelts_list = []
+    @toolset_list   = []
+    @craft_data     = get_data('crafting')['blacksmithing'][@town.capitalize]
+    @equipmanager   = EquipmentManager.new
+
+    @disciplines.each do |name|
+      @toolbelts_list << @settings.send(name + '_belt')
+      @toolset_list   << @settings.send(name + '_tools')
+    end
+
+    unless (repair_info = get_data('town')[@town]['metal_repair']) || args.self_repair
+      DRC.message("No repair info found for #{@town}, exiting")
+      reset_town
+      exit
+    end
+
+    @equipmanager.empty_hands
+
+    if args.drop # dropping off based on remaining args
+      gear_list = args.flex.empty? ? prep_full_repair : build_list(args.flex)
+      UserVars.last_repair_town = @town
+      repair_gear(gear_list, repair_info)
+    elsif args.pickup # by default picks up everything at the last place you dropped off, or your hometown.
+      pickup_repaired_items(repair_info)
+      fput('sort auto head') if @sort_head
+    elsif args.flex.empty? # This represents no flex args, so full repair routine.
+      gear_list = prep_full_repair
+      UserVars.last_repair_town = @town unless args.self_repair
+      repair_gear(gear_list, repair_info, args.self_repair)
+      pickup_repaired_items(repair_info) unless args.self_repair
+      fput('sort auto head') if @sort_head
+    else # Here we have flex args, and no drop/pickup specified, so doing a full routine on the specific items from our flex args.
+      gear_list = build_list(args.flex)
+      UserVars.last_repair_town = @town unless args.self_repair
+      repair_gear(gear_list, repair_info, args.self_repair)
+      pickup_repaired_items(repair_info) unless args.self_repair
+    end
+  end
+
+  def prep_full_repair
+    exit unless should_repair?
+    gear = []
+    @equipmanager.items.each do |item|
+      next if item.skip_repair
+
+      gear << [item.adjective, item.name].join(' ').strip
+    end
+    gear
+  end
+
+  def build_list(selections)
+    gear_list = []
+    @disciplines.each do |toolset|
+      if selections.include?(toolset)
+        gear_list << @settings.send(toolset + '_tools')
+        selections -= [toolset]
+      end
+    end
+    (gear_list + selections).flatten
+  end
+
+  def reset_town(leave = false)
+    if UserVars.last_repair_town
+      DRC.message("Last repair town was: #{UserVars.last_repair_town}. Resetting.") if leave
+      UserVars.last_repair_town = nil
+    else
+      DRC.message("Last repair town not defined.")
+    end
+    exit if leave
+  end
+
+  def should_repair?
+    return true if @repair_timer.nil?
+
+    @repair_timer >= (Time.now - UserVars.repair_timer_snap).to_i
```

Comment:
Brain isn't cooperating today, could someone verify this is accurate? I *think* this is saying, if the time difference (in seconds) between the last time you repaired (repair_timer_snap) and the present time, which would be elapsed time. is GREATER THAN your repair timer, return true.

Reply:
This is checking if `@repair_timer` is greater than, or equal to `(Time.now - UserVars.repair_timer_snap).to_i` if it is, it'll return `true`.